oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/ElasticsearchPropertyIndexTest.java

```diff
@@ -34,29 +34,33 @@
 import org.apache.jackrabbit.oak.spi.state.NodeBuilder;
 import org.apache.jackrabbit.oak.spi.state.NodeStore;
 import org.junit.Assert;
-import org.junit.Rule;
+import org.junit.ClassRule;
 import org.junit.Test;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
 
 import java.util.Arrays;
-import java.util.Collections;
-import java.util.Set;
 
-import static com.google.common.collect.ImmutableSet.of;
-import static org.apache.derby.vti.XmlVTI.asList;
+import static java.util.Collections.singletonList;
 import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.INDEX_DEFINITIONS_NAME;
 import static org.apache.jackrabbit.oak.plugins.index.search.FulltextIndexConstants.PROPDEF_PROP_NODE_NAME;
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.MatcherAssert.assertThat;
 
 public class ElasticsearchPropertyIndexTest extends AbstractQueryTest {
-    @Rule
-    public ElasticsearchManagementRule esMgmt = new ElasticsearchManagementRule();
+    @ClassRule
+    public static final ElasticsearchContainer ELASTIC =
+            new ElasticsearchContainer("docker.elastic.co/elasticsearch/elasticsearch:7.6.0");
```

Comment:
@fabriziofortino 
This would expect docker service to be available on the instance running this test - correct ? Might be problematic in case of windows (I face a lot of issues while running docker on windows).

Maybe we should have an option where this test can be run with an external elastic server as well ? Maybe system properties - if they are present, they take precedence otherwise the docker container is used. wdyt ?

Reply:
To do that we should remove the `ClassRule` and manage the lifecycle for each test. Out of curiosity: what kind of issues do you have running it on docker for windows?

Reply:
>To do that we should remove the ClassRule and manage the lifecycle for each test
Yes, but I understand that it would be a bit of an over work for a test - maybe we can let it be the way it is currently and add a TODO to maybe have a look at it later. 


>what kind of issues do you have running it on docker for windows?
For instance, I had configured it as a startup service, but somehow it never did. And even after manually enabling the service, It used to get stopped every now and then..I gave up trying to fix it after a while and moved on to a linux VM for docker related stuff. (But this was a bit back, probably there could be better support for windows now)